ArcGISRuntimeSDKQt_QMLSamples/Routing/DisplayRouteLayer/DisplayRouteLayer.qml

```diff
@@ -0,0 +1,156 @@
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// [Legal]
+
+import QtQuick 2.12
+import Esri.ArcGISRuntime 100.15
+import QtQuick.Controls 2.12
+import QtQuick.Layouts 1.11
+
+Rectangle {
+    id: rootRectangle
+    clip: true
+    width: 800
+    height: 600
+
+    property var featureCollection: null
+    property var featureCollectionLayer: null
+    property string directions: ""
+
+    MapView {
+        id: mapView
+        anchors.fill: parent
+
+        Component.onCompleted: {
+            // Set and keep the focus on MapView to enable keyboard navigation
+            forceActiveFocus();
+        }
+
+        PortalItem {
+            id: portalItem
+            itemId: "0e3c8e86b4544274b45ecb61c9f41336"
+            Component.onCompleted: load();
+            onLoadStatusChanged: {
+                if (loadStatus !== Enums.LoadStatusLoaded) {
+                    return;
+                }
+
+                // if the portal item is a feature collection, add the feature collection to the map's operational layers
+                if (type === Enums.PortalItemTypeFeatureCollection) {
+                    featureCollection = ArcGISRuntimeEnvironment.createObject("FeatureCollection", {item: portalItem});
+                    featureCollectionLayer = ArcGISRuntimeEnvironment.createObject("FeatureCollectionLayer", {featureCollection: featureCollection});
+                    map.operationalLayers.append(featureCollectionLayer);
+                } else {
+                    console.warn("Portal item with ID '" + itemId + "' is not a feature collection.")
+                }
+            }
+        }
+
+        Map {
+            id: map
+            initBasemapStyle: Enums.BasemapStyleArcGISTopographic
+            initialViewpoint: viewpoint
+        }
+    }
+    ViewpointCenter {
+        id: viewpoint
+        // Specify the center Point
+        center: Point {
+            x: -122.8309
+            y: 45.2281
+            spatialReference: SpatialReference { wkid: 4326 }
+        }
+        // Specify the scale
+        targetScale: 800000
+    }
+    Button {
+        text: "Directions"
+        anchors.bottom: parent.bottom
+        anchors.margins: 40
+        enabled: featureCollection ? featureCollection.loadStatus === Enums.LoadStatusLoaded : false
+        anchors.horizontalCenter: parent.horizontalCenter
+        onClicked: {
+            popup.open();
+            if (popup.opened)
+            {
+                const tables = featureCollection.tables;
+                featureCollection.tables.forEach((table) => {
+                                                     if (table.displayName === "DirectionPoints")
```

Comment:
This should be manually de-indented. Ctrl+i doesn't work well with blocks like `forEach`